lib/src/components/NxColorPicker/__tests__/NxColorPicker.test.tsx

```diff
@@ -4,154 +4,130 @@
 
-    inputs2.forEach(function(input) {
-      expect(input).toHaveProp('name', name2);
-    });
+    for (const selectableColor of selectableColors) {
+      expect(view.getByRole('radio', { name: new RegExp(`My Color Picker ${selectableColor}`, 'i') }))
+          .toBeInTheDocument();
+    }
   });
 
-  it('sets the corresponding color value on each input', function() {
-    const component = getMounted(),
-        labels = component.find('.nx-color-picker__color');
+  it('sets the checked attr to true on the radio matching the value prop', function() {
+    const noneSelectedView = quickRender(),
+        turquoiseSelectedView = quickRender({ value: 'turquoise' });
 
-    expect(labels.filterWhere(label => label.hasClass('nx-selectable-color--sky')).find('input'))
-        .toHaveProp('value', 'sky');
+    expect(noneSelectedView.queryByRole('radio', { checked: true })).not.toBeInTheDocument();
 
-    expect(labels.filterWhere(label => label.hasClass('nx-selectable-color--turquoise')).find('input'))
-        .toHaveProp('value', 'turquoise');
+    expect(turquoiseSelectedView.queryByRole('radio', { checked: true })).toHaveAccessibleName(/turquoise/i);
   });
 
-  it('sets the checked attr to true only on the input matching the value', function() {
-    const noneSelectedComponent = getMounted(),
-        turquoiseSelectedComponent = getMounted({ value: 'turquoise' });
+  it('fires its onChange handler with the color of the clicked radio', async function() {
+    const user = userEvent.setup(),
+        onChange = jest.fn(),
+        view = quickRender({ onChange }),
+        radio = view.getByRole('radio', { name: /turquoise/i });
 
-    expect(noneSelectedComponent.find('input').filterWhere(input => !!input.prop('checked'))).not.toExist();
+    expect(onChange).not.toHaveBeenCalled();
 
-    expect(turquoiseSelectedComponent.find('input').filterWhere(input => !!input.prop('checked')))
-        .toHaveProp('value', 'turquoise');
-  });
+    await user.click(radio);
 
-  it('sets the selected class on the label of the selected color', function() {
-    expect(getMounted({ value: 'turquoise' }).find('.nx-color-picker__color.selected'))
-        .toHaveClassName('nx-selectable-color--turquoise');
+    expect(onChange).toHaveBeenCalledWith('turquoise');
   });
 
-  it('fires its onChange handler with the color of the clicked input', function() {
-    const onChange = jest.fn(),
-        component = getMounted({ onChange });
-
-    expect(onChange).not.toHaveBeenCalled();
+  it('does nothing when an input is clicked with no onChange prop', async function() {
+    const user = userEvent.setup(),
+        view = quickRender(),
+        radio = view.getByRole('radio', { name: /turquoise/i });
 
-    component.find('input').filterWhere(input => input.prop('value') === 'turquoise').simulate('change');
-
-    expect(onChange).toHaveBeenCalledWith('turquoise');
+    await user.click(radio);
   });
```

Comment:
Is this intended to have no expect?

Reply:
> Is this intended to have no expect?

Yes, it simply tests that no exception is thrown

Reply:
Out of curiosity, what might be the exception if there is no expectation?

Reply:
> Out of curiosity, what might be the exception if there is no expectation?

The purpose of this test is just to verify that `onChange` is in fact optional and that the component doesn't blow up if you don't specify it.

Reply:
I see, thanks for the explanation.